communities.md

```diff
@@ -100,3 +100,85 @@ Example: not available
 The supported parameters are:
 * **(mandatory)** parent, the UUID of the parent community
 * page, size [see pagination](README.md#Pagination)
+
+## Creating communities
+
+### Creating top level community
+
+**POST /api/core/communities**
+
+To create a top level community, perform as post with the JSON below to the communities endpoint when logged in as admin.
+
+```
+{
+    "name": "test creation",
+    "metadata": [
+        {
+            "key": "dc.title",
+            "value": "test creation",
+            "language": null
+        }
+    ]
+}
+```
+
+### Creating subCommunity
+
+**POST /api/core/communities**
+
+To create a sub level community, perform as post with the JSON below to the communities endpoint when logged in as admin.
+
+```
+{
+    "name": "test subcommunity",
+    "owningCommunity": "b8872eba-1a79-4b8b-a8f6-55fa8f73197b",
+    "metadata": [
+        {
+            "key": "dc.title",
+            "value": "test subcommunity",
+            "language": null
+        }
+    ]
+}
+```
+
+
+## Updating a community
+
+**PUT /api/core/communities/<:uuid>**
+
+Provide updated metadata information about a specific community, when the update is completed the updated object will be returned. The JSON to update can be found below.
+```
+{
+    "id": "b8872eba-1a79-4b8b-a8f6-55fa8f73197b",
```

Comment:
what we expect about readonly attributes? which error code will be used if the id or uuid is not consistent with the request or if there is an attempt to modify another readonly attribute? should we just reject payloads used to write (POST/PUT) that include readonly attributes? this could be probably easier than validate and can be done at the deserializer layer (and probably it is supported out-of-box by jackson)

Reply:
This is good question.
The reason to have the readonly attributes in here at the moment is to avoid the UI layer needs to know which attributes are readonly. The question here is where the logic should reside on determining what's readonly. By refusing readonly attributes to be specified, the logic should be implemented in both REST and the UI

Reply:
Ok, please specify that update of readonly attributes lead to a 422 UNPROCESSABLE ENTITY error